src/tooltips.cpp

```diff
@@ -60,16 +60,50 @@ void tooltip::init_label()
 	rect game_canvas = video::game_canvas();
 	unsigned int border = 10;
 
+	bool mydebug=false;
+
+	rect huge;
+	huge.h=1000000;
+	huge.w=1000000;
+
 	label.set_font_size(font_size);
 	label.set_color(font::NORMAL_COLOR);
-	label.set_clip_rect(game_canvas);
-	label.set_width(text_width);
+	label.set_clip_rect(huge);
+	label.set_width(text_width);   // If tooltip will be too tall for game_canvas, this could be scaled up appropriately
 	label.set_alignment(font::LEFT_ALIGN);
 	label.set_bg_color(bgcolor);
 	label.set_border_size(border);
 
 	label.create_texture();
 
+	point lsize = label.get_draw_size();
+	int new_text_width = text_width;
+	if(lsize.y > game_canvas.h*0.95) {
```

Comment:
What does the 0.95 mean?

Reply:
When the texture height matches the screen height, the text is sometimes written so close to the edge that it appears to be slightly truncated.   I don't know if this is because I run a slightly odd resolution (1920x1010 due to the xcfe panel), or because create_texture will actually return a rectangle that is just a little taller than the clipped size it is provided in some cases.